client/pages/AddBike.tsx

```diff
@@ -0,0 +1,243 @@
+import React, {useState, useEffect} from 'react';
+import { 
+    KeyboardAvoidingView, 
+    StyleSheet, 
+    Text, 
+    TextInput, 
+    View,
+    TouchableOpacity,
+    Button,
+    Image
+ } from 'react-native';
+import DropDownPicker from 'react-native-dropdown-picker';
+import * as ImagePicker from 'expo-image-picker';
+import * as Location from 'expo-location';
+import tagData from '../constants/tags';
+import { addBike, uploadImage } from '../services/bikes';
+import 'react-native-get-random-values'; // must come before uuid import below
+import { v4 as uuidv4 } from 'uuid';
+
+const AddBike = ({route, navigation}) => {
+
+    const { first_name, user_id } = route.params
+
+    const [errorMsg, setErrorMsg] = useState(null);
+    const [isLoading, setIsLoading] = useState(false);
+    const [loaded, setIsLoaded] = useState(false);
+    const [errorLoading, setErrorLoading] = useState(false);
+
+    const [name, setName] = useState('');
+    const [description, setDescription] = useState('');
+    const [lockCombo, setLockCombo] = useState('');
+    const [tags, setTags] = useState([]);
+    const [location, setLocation] = useState(null);
+
+    // for dropdown picker
+    const [open, setOpen] = useState(false);
+    const [items, setItems] = useState(tagData);
+
+    // for image picker
+    const [image, setImage] = useState(null);
+
+    const pickImage = async () => {
+        // No permissions request is necessary for launching the image library
+        let result = await ImagePicker.launchImageLibraryAsync({
+        mediaTypes: ImagePicker.MediaTypeOptions.All,
+        allowsEditing: true,
+        aspect: [4, 3],
+        quality: 0,
+        });
+        console.log(result);
+        if (!result.canceled) {
+            setImage(result.assets[0].uri);
+            }
+    };
+
+    const handleAddBike = async () => {
+        setIsLoading(true);
+        const imgId = uuidv4(); // ⇨ '9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d'
+        const body = {
+            name: name,
+            description: description,
+            lock_combo: lockCombo,
+            tags: tags,
+            owner: user_id,
+            photo: imgId,
+            release: false,
+            location: {
+                latitude: location.coords.latitude,
+                longitude: location.coords.longitude
+            }
+        };
+        uploadImage(image, imgId)
+            .then(() => {
+                addBike(body).then((response) => {
+                    if (response.status === 201) {
+                        setIsLoading(false);
```

Comment:
Are both setIsLoading and setIsLoaded needed?

Reply:
I think so! This page has three possible ways of being rendered: the add bike form if _isLoading_ is false, a loading screen if _isLoaded_ is true, and a success screen if _isLoading_ is false and _isLoaded_ is true. Would it make sense to differentiate the variable names so it's a little clearer?